Spec/Auth.swift

```diff
@@ -845,6 +845,37 @@ class Auth : QuickSpec {
             }
         }
 
+        // RSA8f3
+        it("ensure the message published with a wildcard '*' does not have a clientId") {
+            let options = AblyTests.commonAppSetup()
+            // Request a token with a wildcard '*' value clientId
+            options.token = getTestToken(clientId: "*")
+            let rest = ARTRest(options: options)
+            rest.httpExecutor = mockExecutor
+            let channel = rest.channels.get("test")
+
+            waitUntil(timeout: testTimeout) { done in
+                let message = ARTMessage(name: nil, data: "no client")
+                expect(message.clientId).to(beNil())
+                channel.publish([message]) { error in
+                    expect(error).to(beNil())
+                    switch extractBodyAsMessages(mockExecutor.requests.first) {
+                    case .Failure(let error):
+                        fail(error)
+                    case .Success(let httpBody):
+                        expect(httpBody.unbox.first!["clientId"]).to(beNil())
+                    }
+                    channel.history { page, error in
+                        expect(error).to(beNil())
+                        expect(page!.items).to(haveCount(1))
+                        expect((page!.items[0] as! ARTMessage).clientId).to(beNil())
+                        done()
+                    }
+                }
+            }
+            expect(rest.auth.clientId).to(beNil())
```

Comment:
The code is wrong, yes. I don't really think there's a conflict between RSA8f3 and RSA12. RSA12a has as precondition "The clientId attribute of a TokenRequest or TokenDetails used for authentication is null, or ConnectionDetails#clientId is null following a connection to Ably", which doesn't apply here.